src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,520 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + RC_KINDS_TYPES
+module stdlib_stats_distribution_uniform
+    use stdlib_kinds
+    use stdlib_error, only : error_stop
+    use stdlib_stats_distribution_PRNG, only : dist_rand
+
+    implicit none
+    private
+
+    real(dp), parameter  :: MESENNE_NUMBER = 1.0_dp / (2.0_dp ** 53 - 1.0_dp)
+    integer(int64), parameter :: INT_ONE = 1_int64
+
+    public :: uniform_distribution_rvs
+    public :: uniform_distribution_pdf
+    public :: uniform_distribution_cdf
+    public :: shuffle
```

Comment:
Should `shuffle` perhaps have a more specific name like `shuffle_uniform`? In case we expect them in other distribution modules as well.

Reply:
the `shuffle` procedure is a kind of sorting, which might be in sorting module. It can be moved to suitable module once `stdlib` is finalized.

Reply:
Okay, let's keep it here and as is then, and if needed we can move it in a future PR.